cmd/oras/internal/option/spec.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package option
+
+import (
+	"fmt"
+
+	"github.com/spf13/pflag"
+	"oras.land/oras/internal/registry"
+)
+
+// ImageSpec option struct.
+type ImageSpec struct {
+	ManifestSupportState registry.ManifestSupportState
+
+	// should be provided in form of `<version>-<manifest type>`
+	specFlag string
+}
+
+// Parse parses flags into the option.
+func (opts *ImageSpec) Parse() error {
+	switch opts.specFlag {
+	case "":
+		opts.ManifestSupportState = registry.ManifestSupportUnknown
+	case "v1.1-image":
+		opts.ManifestSupportState = registry.OCIImage
+	case "v1.1-artifact":
+		opts.ManifestSupportState = registry.OCIArtifact
+	default:
+		return fmt.Errorf("unknown image specification flag: %q", opts.specFlag)
+	}
+	return nil
+}
+
+// ApplyFlags applies flags to a command flag set.
+func (opts *ImageSpec) ApplyFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&opts.specFlag, "image-spec", "", "set OCI image spec version and request manifest type. E.g. `v1.1-image,v1.1-artifact`")
+}
+
+// DistributionSpec option struct.
+type DistributionSpec struct {
+	ReferrersApiSupportState registry.ReferrersApiSupportState
```

Comment:
Having a `ReferrersApiSupportState` seems to heavy and loose for the `DistributionSpec`.

Consider
```suggestion
	// ReferrersAPI indicates the preference of the implementation of the Referrers API. 
	// Set to true for referrers API, false for referrers tag scheme, and nil for auto fallback.
	ReferrersAPI *bool
```